detectors/node/opentelemetry-resource-detector-gcp/src/detectors/GcpDetector.ts

```diff
@@ -44,7 +45,10 @@ import {
  */
 class GcpDetector implements DetectorSync {
```

Comment:
I just saw that this depends on `@opentelemetry/resource@^1.0.0` but the `DetectorSync` was added in a later version. I think we may need to update the package to the version where `DetectorSync` was first added to avoid compile issues. :grimacing: